Products/ZenModel/Device.py

```diff
@@ -1887,7 +1888,11 @@ def renameDevice(self, newId=None, REQUEST=None, retainGraphData=False):
         if newId == '' or newId == oldId:
             return path
 
-        device = self.dmd.Devices.findDeviceByIdExact(newId)
```

Comment:
You still want to do a find because we are looking for a device with the same id regardless of path